src/main/java/de/tum/in/www1/artemis/service/DragAndDropQuizAnswerConversionService.java

```diff
@@ -131,11 +132,11 @@ private void drawPictureDragItem(Graphics2D graphics, DropLocationCoordinates dr
     }
 
     private void drawDropLocation(DragAndDropSubmittedAnswer dragAndDropSubmittedAnswer, Graphics2D graphics, DropLocation dropLocation,
-            DropLocationCoordinates dropLocationCoordinates, boolean showResult) {
+            DropLocationCoordinates dropLocationCoordinates, boolean showResult, QuizQuestion quizQuestion) {
         graphics.setColor(Color.WHITE);
         graphics.fillRect(dropLocationCoordinates.x, dropLocationCoordinates.y, dropLocationCoordinates.width, dropLocationCoordinates.height);
 
-        if (dropLocation.isDropLocationCorrect(dragAndDropSubmittedAnswer)) {
+        if (((DragAndDropQuestion) quizQuestion).isDropLocationCorrect(dragAndDropSubmittedAnswer, dropLocation)) {
```

Comment:
Since this is only ever called with a `DragAndDropQuestion`, can you change the type in the signature and remove the cast?